rust/worker/src/execution/operators/count_records.rs

```diff
@@ -23,16 +25,21 @@ impl CountRecordsOperator {
 pub(crate) struct CountRecordsInput {
     record_segment_definition: Segment,
     blockfile_provider: BlockfileProvider,
+    // Note: this vector needs to be in the same order as the log
+    // for the counting logic to be correct.
+    log_operation_and_id: Vec<(Operation, String)>,
```

Comment:
I'm confused as to why we pull this datastructure out rather than just keep Chunk<LogRecord> which encapsulates the same information

Reply:
yeah there's no particular reason. I was trying to be cautious of cloning the entire log record by selectively cloning only the id and operation but I noticed that it is behind Arc so will change